toolkit/tools/pkg/imagecustomizerlib/rpmsourcesmounts.go

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/packagerepo/repomanager/rpmrepomanager"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safemount.go"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"gopkg.in/ini.v1"
+)
+
+const (
+	rpmsMountParentDirInChroot = "/_localrpms"
+)
+
+// Used to manage (including cleanup) the mounts required by package installation/update.
+type rpmSourcesMounts struct {
+	rpmsMountParentDir        string
+	rpmsMountParentDirCreated bool
+	mounts                    []*safemount.Mount
+	allReposConfigFilePath    string
+}
+
+func mountRpmSources(buildDir string, imageChroot *safechroot.Chroot, rpmsSources []string,
+	useBaseImageRpmRepos bool,
+) (*rpmSourcesMounts, error) {
+	var err error
+
+	var mounts rpmSourcesMounts
+	err = mounts.mountRpmSourcesHelper(buildDir, imageChroot, rpmsSources, useBaseImageRpmRepos)
```

Comment:
Any reason to not integrate this with the `safechroot.MountPoint` type? afaik that will handle cleanup of the mounts etc.

Reply:
The cleanup of logic of `safechroot.MountPoint` lives in the `safechroot.Chroot` type. And the chroot is created well before the point where we are looking to mount the RPMs directories.